pkgs/development/python-modules/dlib/default.nix

```diff
@@ -20,17 +17,24 @@ buildPythonPackage {
       --replace "pytest==3.8" "pytest"
   '';
 
-  # although AVX can be enabled, we never test with it. Some Hydra machines
-  # fail because of this, however their build results are probably used on hardware
-  # with AVX support.
-  checkPhase = ''
-    ${python.interpreter} nix_run_setup test --no USE_AVX_INSTRUCTIONS
+  dontUseCmakeConfigure = true;
+
+  # Pass CMake flags through to the build script
+  preConfigure = ''
+    setupPyBuildFlags="$setupPyBuildFlags $(sed 's/-D\(\S*\)/--set \1/g' <<< "$cmakeFlags")"
   '';
 
-  setupPyBuildFlags = [
-    "--set USE_SSE4_INSTRUCTIONS=${if sse4Support then "yes" else "no"}"
-    "--set USE_AVX_INSTRUCTIONS=${if avxSupport then "yes" else "no"}"
-  ];
+  doCheck = !(
+    # The tests attempt to use CUDA on the build platform.
+    dlib.cudaSupport
 
-  dontUseCmakeConfigure = true;
-}
+      # although AVX can be enabled, we never test with it. Some Hydra machines
+      # fail because of this, however their build results are probably used on hardware
+      # with AVX support.
+      || dlib.avxSupport
+  );
+
+  checkPhase = ''
+    ${python.interpreter} nix_run_setup test
```

Comment:
Any advantage using `"${python.interpreter}"` over `"python"`?

Reply:
Not sure; it was like this already.
I'm pretty sure we should be using `pytestCheckHook` anyway.

Reply:
When rebasing please be aware that this way of running python tests will be removed in the current staging run and pytestCheckHook should be used